pkg/api/types.go

```diff
@@ -311,3 +327,49 @@ func (p *PullPolicy) Set(v string) error {
 	}
 	return nil
 }
+
+// IsSafeFilename checks if the filename provided does not contain any invalid
+// characters.
+func IsSafeFilename(name string) bool {
```

Comment:
I see we negate twice here, once in the return statement, and later in all usages of `IsSafeFilename`.
Perhaps what we want is a `IsInvalidFilename`?
```go
const invalidChars = `\:;*?"<>|%#$!+{}&[],"'` + "`"

func IsInvalidFilename(name string) bool {
	return strings.ContainsAny(name, invalidChars)
}

// ...

	if IsInvalidFilename(s.SourcePath) || IsInvalidFilename(s.DestinationDir) {
 		return fmt.Errorf("invalid characters in file name: %q", value)
 	}
```